src/software/backend/simulation/physics_ball.cpp

```diff
@@ -0,0 +1,42 @@
+#include "software/backend/simulation/physics_ball.h"
+
+#include "shared/constants.h"
+#include "software/backend/simulation/box2d_util.h"
+
+PhysicsBall::PhysicsBall(std::shared_ptr<b2World> world, const Ball &ball)
+{
+    // All the BodyDef must be defined before the body is created.
+    // Changes made after aren't reflected
+    ball_body_def.type = b2_dynamicBody;
+    ball_body_def.position.Set(ball.position().x(), ball.position().y());
+    ball_body_def.linearVelocity.Set(ball.velocity().x(), ball.velocity().y());
+    // The ball can potentially move relatively quickly, so treating it as a "bullet"
+    // helps prevent tunneling and other collision problems
```

Comment:
Perhaps link to the docs for what a "bullet" means?

Reply:
done